### PR TITLE
Fix/pm 215 fix destructor ethercat

### PR DESCRIPTION
**Pull request by Joris-JW**
Closes PM-215

## Description
In the current hardware-interface the destructor, which terminates the ethercat thread, is placed in the march_hardware_interface. When the constructor fails due to an error the destructor does not get called correctly. this results in a still running, zombie kind of, thread.

## Changes
* removed the destructor of the ethercat to EthercatMaster.cpp
* removed the init() function from the constructor of the march_hardware_interface
* refactored the EthercatMaster

### CODE REVIEW

**Review by RutgerVanBeek — DISMISSED** [review]
Looks good. Of course test on exo before merging.

File: march_hardware/src/EtherCAT/EthercatMaster.cpp [hunk 1]
```diff
@@ -61,37 +71,38 @@ void EthercatMaster::start()
...
+ * Set the found slaves to pre-operational state, configure the slaves and move to safe-operational state. If everything
+ * went good move to operational state.
+ */
+void EthercatMaster::ethercatSlaveInitiation()
+{
+  ROS_INFO("Request per-operational state for all slaves");
   ec_statecheck(0, EC_STATE_PRE_OP, EC_TIMEOUTSTATE * 4);
 
-  for (int i = 0; i < jointListPtr->size(); i++)
+  for (auto& i : *jointListPtr)
```
Comment (Roelemans) [inline march_hardware/src/EtherCAT/EthercatMaster.cpp]:
```suggestion
  for (auto& joint : *jointListPtr)
```

File: march_hardware/src/EtherCAT/EthercatMaster.cpp [hunk 2]
```diff
@@ -61,37 +71,38 @@ void EthercatMaster::start()
...
   {
-    jointListPtr->at(i).initialize(ecatCycleTimems);
+    i.initialize(ecatCycleTimems);
```
Comment (Roelemans) [inline march_hardware/src/EtherCAT/EthercatMaster.cpp]:
```suggestion
    joint.initialize(ecatCycleTimems);
```

**Review by Roelemans — CHANGES_REQUESTED** [review]
Looks good, small suggestion :kissing_heart:. Please test on exo!

**Review by Roelemans — APPROVED** [review]

**Review by RutgerVanBeek — APPROVED** [review]

**Pull request merged**